src/codegen/compile.ml

```diff
@@ -6253,6 +6288,19 @@ module MakeSerialization (Strm : Stream) = struct
             remember_failure get_val ^^
             get_val ^^ store_ptr
           )
+          )
+      | Prim Region ->
+         read_alias env (Prim Region) (fun get_arg_typ on_alloc ->
+          let (set_region, get_region) = new_local env "region" in
+          get_arg_typ ^^
+          compile_eq_const (Int32.neg (Option.get (to_idl_prim (Prim Region)))) ^^
+          E.else_trap_with env "read_alias unexpected idl_typ" ^^ (* FAILS! *)
+          Region.alloc env (compile_unboxed_const 0l) (compile_unboxed_const 0l) (Blob.lit env "") ^^ set_region ^^
+          on_alloc get_region ^^
+          get_region ^^ ReadBuf.read_word32 env get_data_buf ^^ Heap.store_field Region.id_field ^^
+          get_region ^^ ReadBuf.read_word32 env get_data_buf ^^ Heap.store_field Region.page_count_field ^^
+          get_region ^^ with_blob_typ env (read_blob ()) ^^ Heap.store_field Region.vec_pages_field ^^
```

Comment:
Why `with_blob_typ`? This tries to read bytes off the type table, but I don’t think you are explicitly adding an entry for the `blob` type to the type table (in `add_typ`), do you? (I don’t think you should, but then remove the `with_blob_typ` here).